scripts/devel/create_devel_task_generator.py

```diff
@@ -161,11 +161,33 @@ def get_dependencies(pkgs, label, get_dependencies_callback):
     return depend_names
 
 
-def _get_build_dependencies(pkg):
-    return pkg.build_depends + pkg.buildtool_depends
+def _get_build_and_recursive_run_dependencies(pkg, pkgs):
+    depends = pkg.build_depends + pkg.buildtool_depends
+    # include recursive run dependencies on other pkgs in the workspace
+    # if pkg A in the workspace build depends on pkg B in the workspace
+    # then the recursive run dependencies of pkg B need to be installed
+    # in order to build the workspace
+    other_pkgs_by_names = dict([(p.name, p) for p in pkgs if p.name != pkg.name])
+    run_depends_in_pkgs = set([d.name for d in depends if d.name in other_pkgs_by_names])
+    while run_depends_in_pkgs:
+        # pick first element from sorted order to ensure deterministic results
+        pkg_name = sorted(run_depends_in_pkgs).pop(0)
```

Comment:
The variable is currently a `set` in order to avoid duplicates but a set can't be sorted. I am not sure if converting the data structure to a list (which can be sorted) makes sense since then it will require manual logic to deduplicate.